kbfsblock/id.go

```diff
@@ -91,6 +104,37 @@ func MakeTemporaryID() (ID, error) {
 	return ID{h}, nil
 }
 
+// MakeRandomIDInRange generates a random block ID using a CSPRNG, distributing
+// the random variable over the interval [start, end), where the full range is
+// [0, MaxUint64). This corresponds to a normalized representation of the
+// range [kbfshash.RawDefaultHash{}, kbfshash.MaxDefaultHash).
+func MakeRandomIDInRange(start, end uint64) (ID, error) {
+	if start >= end {
+		return ID{}, errors.New("Invalid range for random ID")
+	}
+	rangeSize := float64(end - start)
+	randBuf := make([]byte, 8)
+	err := kbfscrypto.RandRead(randBuf)
+	if err != nil {
+		return ID{}, err
+	}
+	// Generate a random unsigned int. Endianness doesn't matter here because
+	// the bytes are random.
+	randUint := binary.BigEndian.Uint64(randBuf)
+	randFloat := float64(randUint) / float64(math.MaxUint64)
```

Comment:
Yeah, I don't think we have to worry so much about the distribution being completely uniform, since we're using it only for eviction purposes. My main concern is that it's not completely clear that it's not critical to have a uniform distribution, etc., so someone reading the code in isolation would be worried and have to look up where it's called to verify that.

So maybe just change the comment to something like "distributing the random variable approximately uniformly over the interval [start, end]." and add "This shouldn't be used for cryptographic purposes." or something along those lines.

I don't care too much about changing it to take floats or uint64s, but I just thought that floats in [0, 1] were simpler since the only existing caller already has such floats.